Add support for ElasticsearchVersion parameter

kraney opened the pull request:
### What does this PR do?

Adds support for the ElasticsearchVersion parameter to the module and state modules, to allow creation & management of ES 2.3 clusters.
### What issues does this PR fix or reference?

None
### Tests written?

Covered by existing tests.

Commit:
Add support for ElasticsearchVersion parameter

cachedout commented:
Hi @kraney 

Would you mind please having a look here? https://jenkins.saltstack.com/job/PR/job/salt-pr-rs-cent7-n/5399/testReport/junit/unit.states.boto_elasticsearch_domain_test/BotoElasticsearchDomainTestCase/test_present_when_domain_exists/

Thanks. :]

Commit:
Update unit test.

Also, version is reported on status rather than config; update code to reflect that.

Commit:
The API does not actually allow version changes to existing deployments

kraney commented:
As far as I can tell, the test failures that remain are in a completely unrelated area.

cachedout commented:
Cool. Thank you very much as always, @kraney .
